### DISCUSSION TITLE
Using @Field on Parameter Properties

### DISCUSSION THREAD

**Discussion by SamHatoum** [MichalLytek/type-graphql#1555] [General]
Hey 👋 a million thanks to you for this great library. 

So I know I can do this easily enough:

```Typescript
@InputType()
export class Foo {
  @Field()
  public readonly bar: string;
  constructor(foo: string) {
    this.foo = foo;
  }
}
```

However if I want compact code using [Parameter Properties](https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties) like this:

```Typescript
@InputType()
export class Foo {
  constructor(@Field() public readonly foo: string) {}
}
```

It doesn't seem like it's supported, and I see these errors:

```console
TS1239: Unable to resolve signature of parameter decorator when called as an expression.
Argument of type  undefined  is not assignable to parameter of type  string | symbol 
```
```console
TS1271: Decorator function return type is  void | TypedPropertyDescriptor<unknown>  but is expected to be  void  or  any 
Type  TypedPropertyDescriptor<unknown>  is not assignable to type  void 
```

I couldn't find anything to state if Parameter Properties are supported in some other way. Any way to do this currently? Or plans to support? Thank you 🙏

**Comment by MichalLytek** [comment]
See #495

**Reply by SamHatoum** [reply]
Oh yay! I did a thorough search. Wonder why I couldn't find it. Thanks
